lib/input/socket_server.go

```diff
@@ -89,7 +89,6 @@ type SocketServer struct {
 	listener  net.Listener
 	conn      net.PacketConn
 
-	retriesMut   sync.RWMutex
```

Comment:
There's definitely an issue with `hasLocked` not being set to true, but I'm surprised this has been flagged as not having effect. We definitely need a lock here because it's our only back pressure mechanism that ensures the queue of retried messages doesn't swell indefinitely. I'll spend some time playing around with this on a branch.

Reply:
Okay this is definitely a false positive, it's assuming it's ineffective because it looks like a common bug https://staticcheck.io/docs/checks#SA2001 but in our case it's necessary. The `hasLocked` part is a real bug though, it should be:

```go
				if !hasLocked {
					hasLocked = true
					t.retriesMut.RLock()
					defer t.retriesMut.RUnlock()
				}
```

It shouldn't be high impact though as it'll just mean we add some extra read locks during an exhaustive retry loop.

Reply:
Thank you so much Ash for looking into this so quickly! I apologize for my initial remark where I dismissed this as a potential refactoring leftover. I sometimes assume that what I don't understand is probably wrong and I really shouldn't do that. I'm sorry!

I fixed the hasLocked flag issue in 50e461d2 and I re-added the locks and instructed statickcheck to stop warning about them in 74bf1e1e. I got misled by the fact that there is nothing happening between lock and unlock, which I thought will be ineffective. I never implemented back pressure mechanisms before, TBH, and I know they're a pain to get right.

Reply:
No worries, personally I'd rather you lean on the side of deleting code that looks unnecessary as that's usually going to be the correct approach, and when it's not there really ought to be a test failure to catch it anyway.